### PR TITLE
switched to intToDeviceId

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Comment by peterBrxwn** [comment]
[[RN] Fix the default case in "stringToDeviceIDType"](https://trello.com/c/0ADvmwdO/1366-rn-fix-the-default-case-in-stringtodeviceidtype)

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
It seems all okey, but why is this needed?
Is not it about fixing the default case of "stringToDeviceIDType"

Just an idea, maybe, both native sides can return full string. And, RN can do a simple contains check whether or not we have that type and return it. Maybe we can even do not do that. It can return directly full string from both native sides. Why we are using extra switch cases for it. 

Does this putting any value? Or is it beneficial?

**Review by arifBurakDemiray — APPROVED** [review]

File: ios/src/CountlyReactNative.m [hunk 1]
```diff
@@ -401,17 +401,15 @@ + (void)log:(NSString *)theMessage {
 RCT_REMAP_METHOD(getDeviceIDType, getDeviceIDTypeWithResolver : (RCTPromiseResolveBlock)resolve rejecter : (RCTPromiseRejectBlock)reject) {
     dispatch_async(dispatch_get_main_queue(), ^{
         CLYDeviceIDType deviceIDType = [Countly.sharedInstance deviceIDType];
-        NSString *deviceIDTypeString = NULL;
+        NSNumber *deviceIDTypeInt = NULL;
```
Comment (ArtursKadikis) [inline ios/src/CountlyReactNative.m:404]: try NSInteger

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
add a test with the number 1337

File: ios/src/CountlyReactNative.m (hunk 1 above)
Comment (peterBrxwn) [reply ios/src/CountlyReactNative.m]: done

File: ios/src/CountlyReactNative.m [hunk 2]
```diff
@@ -401,17 +401,15 @@ + (void)log:(NSString *)theMessage {
 RCT_REMAP_METHOD(getDeviceIDType, getDeviceIDTypeWithResolver : (RCTPromiseResolveBlock)resolve rejecter : (RCTPromiseRejectBlock)reject) {
     dispatch_async(dispatch_get_main_queue(), ^{
         CLYDeviceIDType deviceIDType = [Countly.sharedInstance deviceIDType];
-        NSString *deviceIDTypeString = NULL;
```
Comment (ArtursKadikis) [inline ios/src/CountlyReactNative.m:404]: Change it back to NSNumber

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

File: Utils.js [hunk 3]
```diff
@@ -9,28 +9,27 @@ const DeviceIdType = {
 
 /**
  *
- * internal countly function that converts String to DeviceIdType.
```
Comment (turtledreams) [inline Utils.js:12]: have log for each case please

File: Countly.js [hunk 4]
```diff
@@ -521,7 +521,7 @@ Countly.getDeviceIDType = async function () {
         L.e("getDeviceIDType, unexpected null value from native side");
```
Comment (turtledreams) [inline Countly.js:521]: It think we can remove this but add a log to default value of intToDeviceID to indicate there was an unexpected value coming from the native side

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

File: __tests__/utils.test.js [hunk 5]
```diff
@@ -1,57 +1,71 @@
-const { stringToDeviceIDType, DeviceIdType } = require("../Utils.js");
+const { intToDeviceIDType, DeviceIdType } = require("../Utils.js");
```
Comment (turtledreams) [inline __tests__/utils.test.js:2]: lets also change the name of the test file to something more specific. lets not put all utils test to same file

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: __tests__/utils.test.js (hunk 5 above)
Comment (peterBrxwn) [reply __tests__/utils.test.js:2]: intToDeviceIDType.test.js?

File: ios/src/CountlyReactNative.m (hunk 2 above)
Comment (peterBrxwn) [reply ios/src/CountlyReactNative.m:404]: done

File: Countly.js (hunk 4 above)
Comment (peterBrxwn) [reply Countly.js:521]: Done

File: Utils.js (hunk 3 above)
Comment (peterBrxwn) [reply Utils.js:12]: done

**Review by turtledreams — APPROVED** [review]

**Review by ArtursKadikis — APPROVED** [review]

**Pull request merged**